skglm/solvers/fista.py

```diff
@@ -27,6 +28,9 @@ class FISTA(BaseSolver):
            https://epubs.siam.org/doi/10.1137/080716542
     """
 
+    _datafit_required_attr = ("get_global_lipschitz", ("gradient", "gradient_scalar"))
+    _penalty_required_attr = (("prox_1d", "prox_vec"),)
```

Comment:
why does FISTA need prox_1D, it is not used in the code below

Reply:
It is used in ``_prox_vec`` which take as argument the penalty (cf. https://github.com/PABannier/skglm/blob/e687bc2ecaacfe920b9aaad3e33e1f0cbdbac683/skglm/solvers/fista.py#L83)

The algorithm works if penalty has either of ``prox_1d`` or  ``prox_vec``. 
(for reference https://github.com/scikit-learn-contrib/skglm/pull/137#discussion_r1613225915)

Reply:
got it, thanks !